sql/_05_plcsql/_01_testspec/_02_declaration/_01_parameter/_03_type_conversion/cases/02_01_03-14_sql_to_param_VARCHAR.sql

```diff
@@ -0,0 +1,210 @@
+
+
+call print_message('t_BIT(8)_VARCHAR. This scenario is a failure.');
+create or replace procedure t_BIT_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_BIT_VARCHAR('BIT(8)', 'VARCHAR', 0xaa ) ;
+drop procedure t_BIT_VARCHAR ;
+
+
+call print_message('t_BIT VARYING_VARCHAR. This scenario is a failure.');
+create or replace procedure t_BITVARYING_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_BITVARYING_VARCHAR('BIT VARYING', 'VARCHAR', 0xaa ) ;
+drop procedure t_BITVARYING_VARCHAR ;
+
+
+call print_message('t_CHAR_VARCHAR. This scenario is a success.');
+create or replace procedure t_CHAR_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_CHAR_VARCHAR('CHAR', 'VARCHAR', 'CHAR CUBRID' ) ;
+drop procedure t_CHAR_VARCHAR ;
+
+
+call print_message('t_VARCHAR_VARCHAR. This scenario is a success.');
+create or replace procedure t_VARCHAR_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_VARCHAR_VARCHAR('VARCHAR', 'VARCHAR', 'VARCHAR CUBRID' ) ;
+drop procedure t_VARCHAR_VARCHAR ;
+--BUG
+
+call print_message('t_SET_VARCHAR. This scenario is a failure.');
+create or replace procedure t_SET_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_SET_VARCHAR('SET', 'VARCHAR', {'c','c','c','b','b','a'} ) ;
+drop procedure t_SET_VARCHAR ;
+--BUG
+
+call print_message('t_MULTISET_VARCHAR. This scenario is a failure.');
+create or replace procedure t_MULTISET_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_MULTISET_VARCHAR('MULTISET', 'VARCHAR', {'c','c','c','b','b','a'} ) ;
+drop procedure t_MULTISET_VARCHAR ;
+--BUG
+
+call print_message('t_LIST_VARCHAR. This scenario is a failure.');
+create or replace procedure t_LIST_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_LIST_VARCHAR('LIST', 'VARCHAR', {'c','c','c','b','b', 'a'} ) ;
+drop procedure t_LIST_VARCHAR ;
+--BUG
+
+call print_message('t_ENUM_VARCHAR. This scenario is a failure.');
```

Comment:
It's a failure, but the answer file it was successful. Need to check.

Reply:
```suggestion
call print_message('t_ENUM_VARCHAR. This scenario is a success.');
```
It is a success as in ENUM to CHAR case

Reply:
해당 케이스는 ENUM타입을 테스트 하는 것이 목적이므로 아래와 같이 변경하였습니다.
ENUM 컬럼타입의 테이블을 만들고, ENUM컬럼을 파라미터로 전달하여 확인하는 방법으로 변경하였습니다.

수행결과는 ENUM 타입은 미지원 함으로 아래와 같은 에러메세지가 출력됩니다.

ERROR: Stored procedure execute error: unsupported argument type - ENUM

call print_message('t_ENUM_VARCHAR. This scenario is a failure.');
create or replace function t_ENUM_VARCHAR(sql_type string, procedure_type string, param VARCHAR ) return varchar(100) as begin
    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param );
    return 'abc';
end;

CREATE TABLE enum_tbl (
    color ENUM ('red', 'yellow', 'blue', 'green')
);
INSERT INTO enum_tbl (color) VALUES ('yellow');

select t_ENUM_VARCHAR( 'ENUM', 'VARCHAR', color ) from enum_tbl;
drop function t_ENUM_VARCHAR ;
drop table enum_tbl;